tests/framework/utils.py

```diff
@@ -462,6 +465,11 @@ def get_process_pidfd(pid):
         pidfd = os.pidfd_open(pid)
     except ProcessLookupError:
         return None
+    except OSError as err:
+        if err.errno == errno.EINVAL:
+            return None
```

Comment:
I think we do want to raise exceptions in this case, because it might be the only thing that tells us that the firecracker process died unexpectedly (e.g. when we call .kill() we expect there to be a process, and if it's already dead that would indicate a bug). Is the intermittent failure just because of race conditions? In that case, can we just create the pidfd before sending the SIGKILL to fix it? :o

Reply:
synced offline, I got this wrong. The check is in `wait_process_termination`, e.g. at a point in time where we only wait for the process to go away. We don't assert it's still alive here, that happens in .kill()